common/persistence/cassandra/mutable_state_store.go

```diff
@@ -478,16 +478,38 @@ func (d *MutableStateStore) CreateWorkflowExecution(
 	return &p.InternalCreateWorkflowExecutionResponse{}, nil
 }
 
+// GetWorkflowExecution gets the workflow execution data from the cassandra DB.
+// If the request contains a runID it loads and returns the execution identified by that run ID.
+// If the request does not contain a runID, it first looks up the current run ID and then loads the corresponding execution object.
 func (d *MutableStateStore) GetWorkflowExecution(
 	ctx context.Context,
 	request *p.GetWorkflowExecutionRequest,
 ) (*p.InternalGetWorkflowExecutionResponse, error) {
+	runID := request.RunID
+	if runID == "" {
+		// get current runid using the permanentRunID record.
+		query := d.Session.Query(templateGetCurrentExecutionQuery,
+			request.ShardID,
+			rowTypeExecution,
+			request.NamespaceID,
+			request.WorkflowID,
+			permanentRunID,
+			defaultVisibilityTimestamp,
+			rowTypeExecutionTaskID,
+		).WithContext(ctx)
```

Comment:
should we also extend the context timeout for this case?

Reply:
I don't think we need to extend the timeout. Currently the application explicitly makes this query and the below query (but separately). They all finish within the deadline.
Once we fix the application to not explicitly load the current runID, this codepath is taken and the orverall number of DB queries within the highlevel API remains the same (or reduces in some instances). Until then this codepath is not hit. So I think we are fine without extending the deadline.